test-framework/security-jwt/src/main/java/io/quarkus/test/security/jwt/JwtTestSecurityIdentityAugmentor.java

```diff
@@ -0,0 +1,183 @@
+                return Json.createValue((Long) convertedClaimValue);
+            } else if (convertedClaimValue instanceof Integer) {
+                return Json.createValue((Integer) convertedClaimValue);
+            } else if (convertedClaimValue instanceof Boolean) {
+                return (Boolean) convertedClaimValue ? JsonValue.TRUE : JsonValue.FALSE;
+            }
+        }
+        return convertedClaimValue;
+    }
+
+    protected Claims getClaimType(String claimName) {
+        Claims claimType;
+        try {
+            claimType = Claims.valueOf(claimName);
+        } catch (IllegalArgumentException e) {
+            claimType = Claims.UNKNOWN;
+        }
+        return claimType;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T convertClaimValue(Claim claim) {
+        if (claim.type() != Claim.Type.UNKNOWN) {
+            Converter<?> converter = converters.get(claim.type());
+            if (converter != null) {
+                return (T) converter.convert(claim.value());
+            } else {
+                throw new RuntimeException("Unsupported claim type: " + claim.type().name());
+            }
+        } else if (standardClaimConverteres.containsKey(claim.key())) {
+            Converter<?> converter = standardClaimConverteres.get(claim.key());
+            return (T) converter.convert(claim.value());
+        } else {
+            return (T) claim.value();
+        }
+    }
+
+    private static interface Converter<T> {
+        T convert(String value);
+    }
+
+    private static class StringConverter implements Converter<String> {
+        @Override
+        public String convert(String value) {
+            return value;
+        }
+    }
+
+    private static class IntegerConverter implements Converter<Integer> {
+        @Override
+        public Integer convert(String value) {
+            return Integer.valueOf(value);
+        }
+    }
+
+    private static class LongConverter implements Converter<Long> {
+        @Override
+        public Long convert(String value) {
+            return Long.valueOf(value);
+        }
+    }
+
+    private static class BooleanConverter implements Converter<Boolean> {
+        @Override
+        public Boolean convert(String value) {
+            return Boolean.valueOf(value);
+        }
+    }
+
+    private static class JsonObjectConverter implements Converter<JsonObject> {
+        @Override
+        public JsonObject convert(String value) {
+            try (JsonReader jsonReader = Json.createReader(new StringReader(value))) {
+                return jsonReader.readObject();
+            }
+        }
+    }
+
+    private static class JsonArrayConverter implements Converter<JsonArray> {
+        @Override
+        public JsonArray convert(String value) {
+            try (JsonReader jsonReader = Json.createReader(new StringReader(value))) {
+                return jsonReader.readArray();
+            }
+        }
+    }
```

Comment:
Hint: you can encapsulate this as methods in the enum :)

Reply:
Yeah :-), I'll give a try

Reply:
I'll probably need to move it to its own file to make a bit cleaner. Will deal with it tomorrow as I have to sign off, thanks :+1:

Reply:
Hey @gastaldi, @geoand I spent some time this morning and I could not find a neat way to do it. I have a few num implementations in `smallrye-jwt` which their own custom methods, and possibly in OIDC, but I found with enums implementing generic interfaces it was getting a bit messy,  example,
```
// This one which is currently used is not easy to apply  to enum
interface Converter<T> {
   T convert(String value);
}

// so:

interface Converter2 {
   <T> T convert(String value);
}

// and then:
enum Type implements Converter {
   LONG,
   STRING;
   // Suppress the warning here
   @Overrides
   T convert(String value) {
       if (LONG.name().equals(name()) {
            return (T) Long.valueOf(value); 
       } 
       // etc
   }
}
```

and I found it a bit messy. Various suggestions online with how to make it work but none looks great. With a non-generic interface it is straightforward.

What do you think ?
thanks

Reply:
I don't think we need generics here, so that's fine

Reply:
Also I don't think you need the `Converter` interface if you are defining the `convert` method in the enum

Reply:
@gastaldi Sure.
What looks a bit not ideal to me is that I have, right now, the map linking well-known claims to converters, for ex, `exp` -> `LongConverter()`, with `@Claim(key="exp", value="1234")` requiring no type hint. If we move it to the enum, then even when I discover that `Type.LONG` is what should convert it, and I do `Type.LONG.convert("1234")` then the convert implementation would still have to go through if/else checks to figure out how to convert, it does not know that the only think it has to do is Long.valueOf("1234"), see what I mean ?

Reply:
I'm totally happy/open to try to convert, just does not seem to work very neatly

Reply:
@gastaldi Looks great, did not think of returning Object, thanks

Reply:
@gastaldi OK, just added a commit, the simplifications are massive, definitely was worse pursuing :-), thanks.

Does it look OK to you or anything else you'd like to change ?

Reply:
Just added a minor comment. But it looks fine to me